release_draft.md

```diff
@@ -0,0 +1,299 @@
+
+  1. It is required to clone the corresponding release repository to update the
+  binary version:
+
+  ```bash
+  # Ignition library named foo and major version X
+  git clone https://github.com/ignition-release/ign-fooX-release
+  ```
+
+  2. To bump the package versions that will appear in Debian/Ubuntu binary packages
+  there is a helper script in `release-tools` (see
+  [prerequisites](#prerequisites)). The script is called `changelog_spawn.sh` and
+  require to be executed while the active directory is a `release repository`:
+
+  ```bash
+  # Ignition library named foo and major version X
+  cd ign-fooX-release
+  ~/release-tools/release-repo-scripts/changelog_spawn.sh X.Y.Z-R
+
+  # Example ign-cmake2 bumped from 2.0.0 to 2.0.1
+  cd ign-cmake2-release
+  ~/release-tools/release-repo-scripts/changelog_spawn.sh 2.0.1-1
+  ```
+
+  `changelog_spawn.sh` will display information about the Ubuntu/Debian versions
+  being updated as well as a `git diff` before uploading information to the GitHub
+  release repository.
+
+## Launch the release in the building server
+
+After updating the code and releasing metadata everything is ready to launch the
+build in the server. Now, the following needs to happen:
+
+ 1. Generate a tarball with the Ignition library sources corresponding to the new
+    version in the Ignition developer local system. Upload the tarball to S3 cloud storage.
+ 1. Request `build.osrfoundation.org` server to start the jobs for:
+     1. Debian/Ubuntu: use `ign-fooX-debbuilder` job names
+     1. Brew: entry job is `generic-release-homebrew_pull_request_updater`
+
+The `release.py` script will perform all these actions.
+
+### Executing release.py
+
+Make sure you are in the source code repository before running `release.py`. You should be on the branch to be released, after the pull request bumping the version has been merged (run `git status` to check the branch, and `git log` to check that the version bump pull request has been included). Running `release.py` from the source code repository will generate and upload some Git tags ("release tags") to the source code repository.
+
+You will also need the token described in the [credentials section](#credentials).
+
+#### dry-run simulation mode
+
+The `release.py` tool supports a `--dry-run` flag that allows users to simulate releases
+(nothing is modified) in order to ensure that the correct arguments are being used to trigger a particular release.
+Ignition releasers should **always** call `release.py` with `--dry-run` first in order to ensure that proper commands are being used to trigger releases.
+
+The script needs to be run from the repository with the source code (i.e., the repository where the Ignition library version bump pull request took place):
+
+```bash
+# Example of dry-run for ign-cmake2 bumped to 2.0.1
+cd ign-cmake2
+git checkout ign-cmake2
+~/release-tools/release-repo-scripts/release.py ign-cmake2 2.0.1 <test_token> --dry-run
```

Comment:
I had a hard time undertanding what is the `<test_token>`. Maybe we should add a note about this, how to get it, who can provide this, etc.

Reply:
2a4f112